### PR TITLE
Removed the need for validator-address config param

### PR DESCRIPTION
**Pull request by Maharacha**
In the first implementation of this, we choose to use a config param because it would take too long time to the `update-status` hook to find the address. However, the trade off with that is that we need to rely on that the correct address is set in the config. I did a benchmark to see how long time it actually takes. On Kusama, which has 1000 validators it took 20 seconds to do the lookup. That is worst case if the node is not validating so that it has to check all the 1000 addresses. Polkadot has 297 validators, and the parachains usually only have a few collators. So I think we can actually go with this, which is a much more secure way of knowing if a node is validating or not. And it will only do this lookup when `--validator/--collator` is in the service args.

* `update-status` does not show if the validator will validate next era since it requires a validator address as argument. The action is still there though but with a required param added to it. 
* `find-validator-address` action is still there since `update-status` only shows `yes/no`. With the action you get both address and session key.

### CODE REVIEW

**Comment by Maharacha** [comment]
I have only tested this locally yet. I will test it on at least one validator and one collator in prod as well.

File: src/charm.py [hunk 1]
```diff
@@ -139,18 +138,16 @@ def _on_update_status(self, event: ops.UpdateStatusEvent) -> None:
...
             for i in range(connection_attempts):
                 time.sleep(5)
                 try:
                     is_syncing = str(PolkadotRpcWrapper(rpc_port).is_syncing())
                     status_message = f'Syncing: {is_syncing}'
-                    address = self.config.get('validator-address')
-                    if address:
-                        if PolkadotRpcWrapper(rpc_port).is_validating_this_era(address):
+                    if service_args.is_validator:
+                        if PolkadotRpcWrapper(rpc_port).is_validating_this_era():
```
[the remark this answers is not in the document]
Comment (Maharacha) [reply src/charm.py:149]: It's a valid point. I will meditate on that.

**Comment by Maharacha** [comment]
> > find-validator-address action is still there since update-status only shows yes/no. With the action you get both address and session key.
> 
> This action will only return a address if the node is validating this era right? But this was the case before as well just reminding myself on how this works.

Yes, same behavior as before.

**Comment by jakobilobi** [comment]
Please target this PR at `develop` :pray:

File: src/charm.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Maharacha) [reply src/charm.py]: I have meditated and added a boolean for it. Too ugly?

**Comment by jakobilobi** [comment]
> I have only tested this locally yet. I will test it on at least one validator and one collator in prod as well.

Did you do this yet?

If so, let's merge to `develop` and do a revision release soon.

**Comment by Maharacha** [comment]
> > I have only tested this locally yet. I will test it on at least one validator and one collator in prod as well.
> 
> Did you do this yet?
> 
> If so, let's merge to `develop` and do a revision release soon.

Have not tested it yet. Can do tomorrow.

**Comment by Maharacha** [comment]
> > > I have only tested this locally yet. I will test it on at least one validator and one collator in prod as well.
> > 
> > 
> > Did you do this yet?
> > If so, let's merge to `develop` and do a revision release soon.
> 
> Have not tested it yet. Can do tomorrow.

@jakobilobi it's now tested

**Pull request merged**